### PR TITLE
Always save AOVs to EXR files.

### PR DESCRIPTION
**Pull request by est77**
Fix AOV related bugs.

### CODE REVIEW

File: src/appleseed/renderer/modeling/frame/frame.cpp [hunk 1]
```diff
@@ -751,10 +745,10 @@ bool Frame::write_main_image(const char* file_path) const
...
     if (impl->m_denoising_mode == DenoisingMode::WriteOutputs)
     {
-        if (ends_with(lower_case(file_path), ".exr"))
-            impl->m_denoiser_aov->write_images(file_path);
-        else
-            RENDERER_LOG_ERROR("denoiser outputs can only be saved to exr images.");
+        bf::path boost_file_path(file_path);
+        boost_file_path.replace_extension(".exr");
+
+        impl->m_denoiser_aov->write_images(file_path);
```
Comment (dictoon) [inline src/appleseed/renderer/modeling/frame/frame.cpp]: You're not using `boost_file_path` here. Intentional?

File: src/appleseed/renderer/modeling/aov/aov.cpp [hunk 2]
```diff
@@ -163,6 +163,8 @@ void UnfilteredAOV::create_image(
             tile_height,
             num_channels,
             PixelFormatFloat);
+
+    clear_image();
```
Comment (dictoon) [inline src/appleseed/renderer/modeling/aov/aov.cpp]: Whyu does `UnfilteredAOV::create_image()` clears the image but `AOV::create_image()` doesn't? Can we make the behavior more consistent?

Comment (est77) [reply src/appleseed/renderer/modeling/aov/aov.cpp]: There are 2 reasons. One is that for some AOVs, 0 is not the default channel value (normals for example).
The other is that there is an extra hidden channel in in the images of unfiltered AOVs.
The accumulators use it to keep track of the nearest sample to the pixel center seen so far.

We can clear all the images, but it is not needed for the others AOVs and by clearing them we will allocate all the tiles in the image.

Comment (dictoon) [reply src/appleseed/renderer/modeling/aov/aov.cpp]: Thanks for the explanation. Can you add it as a comment in the code?

Comment (est77) [reply src/appleseed/renderer/modeling/aov/aov.cpp]: Fixed and rebased

**Comment by est77** [comment]
Fixed and rebased

**Pull request merged**